contentgrid-spring-querydsl/src/main/java/com/contentgrid/spring/querydsl/predicate/AbstractSimpleQuerydslPredicateFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.contentgrid.spring.querydsl.predicate;
+
+import com.contentgrid.spring.querydsl.annotation.QuerydslPredicateFactory;
+import com.contentgrid.spring.querydsl.mapping.UnsupportedCollectionFilterPredicatePathTypeException;
+import com.querydsl.core.types.Path;
+import com.querydsl.core.types.Predicate;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Base class for "simple" {@link QuerydslPredicateFactory}s.
+ * <p>
+ * A simple predicate factory binds exactly one path the {@link com.contentgrid.spring.querydsl.annotation.CollectionFilterParam} annotation is placed on
+ *
+ * @param <T> The necessary type of path this factory can bind to
+ * @param <S> Type of the values in collection that will be mapped to the path
+ */
+@RequiredArgsConstructor
+public abstract class AbstractSimpleQuerydslPredicateFactory<T extends Path<S>, S> implements QuerydslPredicateFactory<Path<?>, S> {
+    @Override
+    final public Stream<Path<?>> boundPaths(Path<?> path) {
+        return Stream.of(coercePath(path));
+    }
+
+    @Override
+    final public Optional<Predicate> bind(Path<?> path, Collection<? extends S> values) {
+        return bindCoerced(coercePath(path), values);
+    }
+
+    /**
+     * Converts the generic path type that {@link com.contentgrid.spring.querydsl.annotation.CollectionFilterParam} to a specific type that {@link #bindCoerced(Path, Collection)} can work upon
```

Comment:
I'm having trouble parsing this sentence

Reply:
I think I got interrupted while writing that sentence.

Is this better?

```suggestion
     * Converts the generic path type to a specific type that {@link #bindCoerced(Path, Collection)} can work upon
```